src/Microsoft.Performance.SDK.Runtime/ProcessingSourceExecutor.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.Performance.SDK.Runtime
     /// <summary>
     ///     Executes the processor of a data source.
```

Comment:
>a data source [](http://example.com/codeflow?start=38&length=13)

depending on which data source this was referring to, it should at minimum be plural. or perhaps changed to processing source. and honestly, this whole class feels more coupled to a custom data processor than a processing source. i wonder if the name should be DataProcessorExecutor... or DataSourceProcessor... or something less tied to a ProcessingSource

Reply:
I think this is referencing `ProcessingSource`. Can you elaborate on why this should this be plural? It's executing a single CustomDataProcessor that is tied to a single ProcessingSource

As far as renaming the clas: I chatted with @zachnew-msft offline and this entire class might get refactored during later work. I think the class name can stay as-is for now, unless you have strong feelings about renaming it